invenio/modules/oauthclient/contrib/orcid.py

```diff
@@ -127,17 +127,20 @@ def account_info(remote, resp):
 @session_manager
 def account_setup(remote, token):
     """Perform additional setup after user have been logged in."""
-    from invenio.modules.accounts.models import User, UserEXT
-    from invenio.ext.sqlalchemy import db
-    from ..handlers import token_session_key
+    import json
+    import requests
 
     from sqlalchemy.orm.exc import MultipleResultsFound, NoResultFound
 
-    import json
-    import requests
+    from invenio.modules.accounts.models import User, UserEXT
+    from invenio.ext.sqlalchemy import db
+    from ..handlers import token_session_key
 
     orcid = session.get(token_session_key(remote.name) +
-                        "_account_info").get("external_id")
+                        "_account_info", {}).get("external_id")
+
+    if not orcid:
```

Comment:
With the current code, if I understood correctly, this situation will happen when the user has already linked his local account to an external one and then [this condition](https://github.com/inveniosoftware/invenio/blob/pu/invenio/modules/oauthclient/handlers.py#L246) is True

```python
if not token.remote_account.extra_data and \
       remote.name in signup_handlers:
        handlers['setup'](token)
```

which is the case when no ``extra_data`` is available in that table.

I am not sure if it should be considered as an error to be logged or something to ignore, @lnielsen maybe has some input on this?